### PR TITLE
Codesign Lottie.xcframework

### PR DESCRIPTION
**Pull request by calda**
This PR updates the job that builds `Lottie.xcframework` to codesign the bundle using a self-signed certificate. 

When viewing `Lottie.xcframework` in Xcode 15 it now shows the following signature information:

Screenshot 2023-12-14 at 8 10 28 PM

We'll publish the signature fingerprint in the README so consumers can validate they're using an authentic build.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -100,13 +100,18 @@ jobs:
           - '13.4.1' # Swift 5.6.1
     steps:
       - uses: actions/checkout@v2
+      - uses: apple-actions/import-codesign-certs@v2
+        with:
+          p12-file-base64: ${{ secrets.SIGNING_CERTIFICATE_BASE_64 }}
+          p12-password: ${{ secrets.SIGNING_CERTIFICATE_PASSWORD }}
```
Comment (calda) [inline .github/workflows/main.yml:106]: I followed these steps to create a self-signing certificate and installed it as a secret in this repository:

 - Create code signing certificate in Keychain Assistant:
    - First you have to create a custom Certificate Authority in order to issue a certificate that can be used for code signing. Follow these instructions: https://www.simplified.guide/macos/keychain-ca-code-signing-create
    - Then you can create a certificate for code signing. Follow these instructions: https://www.simplified.guide/macos/keychain-cert-code-signing-create
- Export .p12 file for both the code signing certificate and the root certificate authority, protected with a password.
- Add certificate password as GitHub actions secret (`SIGNING_CERTIFICATE_PASSWORD`)
- Convert certificate file to base64, e.g. using `base64 -i "Lottie iOS Code Signing.p12"`.
- Added base 64 as secret to GitHub repo, e.g. using `SIGNING_CERTIFICATE_BASE_64` key

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -86,11 +86,15 @@ jobs:
           - '14.1' # Swift 5.7.1
     steps:
       - uses: actions/checkout@v2
+      - uses: apple-actions/import-codesign-certs@v2
+        with:
+          p12-file-base64: ${{ secrets.SIGNING_CERTIFICATE_BASE_64 }}
```
Comment (jparise) [inline .github/workflows/main.yml:91]: nit: `_BASE64` would be a little more natural spelling, but no strong feelings on that.

File: README.md [hunk 3]
```diff
@@ -85,6 +85,14 @@ Lottie supports Swift / Xcode versions back to the minimum version that is permi
 
 Lottie does not collect any data. We provide this notice to help you fill out [App Privacy Details](https://developer.apple.com/app-store/app-privacy-details/). We additionally provide a [privacy manifest](https://github.com/airbnb/lottie-ios/blob/master/PrivacyInfo.xcprivacy) which can be included in your app.
 
+## Security
+
+We distribute XCFramework bundles for each release on [GitHub](https://github.com/airbnb/lottie-ios/releases/latest). In Lottie 4.4.0 and later, these XCFramework bundles include a [code signature](https://developer.apple.com/documentation/xcode/verifying-the-origin-of-your-xcframeworks). These bundles are self-signed under the name "Lottie iOS" and have the following fingerprint:
```
Comment (jparise) [inline README.md:90]: Should we include some instructions on how someone could use the fingerprint to verify the integrity of their framework artifact?

Comment (calda) [reply README.md:90]: Good idea, updated: https://github.com/airbnb/lottie-ios/pull/2259/commits/c7944f61dfd7300ca9d735df9b00c4dda0e7c739

I couldn't easily find out how to verify this outside of Xcode, e.g. via the command line. Open to suggestions if you have any ideas! 

I tried ` codesign -dv --verbose=4 Lottie.xcframework`, but it doesn't show this specific fingerprint string:

```
Executable=/Users/calstephens/Downloads/BuildProducts/Lottie.xcframework/Info.plist
Identifier=Lottie
Format=bundle
CodeDirectory v=20100 size=183 flags=0x0(none) hashes=1+3 location=embedded
Hash type=sha256 size=32
CandidateCDHash sha1=8e6caf486f23b91f5bbc5eb92c40343ca894cf0d
CandidateCDHashFull sha1=8e6caf486f23b91f5bbc5eb92c40343ca894cf0d
CandidateCDHash sha256=99b40b7bf0ff70ea7abab9e549d3bdd01b26226b
CandidateCDHashFull sha256=99b40b7bf0ff70ea7abab9e549d3bdd01b26226b613be5bca5976e32834aa4f7
Hash choices=sha1,sha256
CMSDigest=4baf8f8e561d838359115d9a23741afeac1a999720d6715a10c5b0238a988081
CMSDigestType=2
Page size=none
CDHash=99b40b7bf0ff70ea7abab9e549d3bdd01b26226b
Signature size=6332
Authority=Lottie iOS
Timestamp=Dec 15, 2023 at 9:15:20 AM
Info.plist entries=3
TeamIdentifier=not set
Sealed Resources version=2 rules=10 files=83
Internal requirements count=1 size=84
```

Perhaps one of the hashes here corresponds to the fingerprint shown in Xcode via some encoding, but I'm not sure.

**Review by jparise — APPROVED** [review]

**Pull request merged**